### PR TITLE
Refactor runner configuration, add labels to dataflow options

### PR DESCRIPTION
**Pull request by zhilingc**
**What this PR does / why we need it**:

Adds option of setting labels for dataflow jobs, and cleans up the job configuration code a bit, also ties the runner configuration classes to the protos. Both runners extend the `RunnerConfig` class which support conversion of the config fields to beam args.

### CODE REVIEW

File: core/src/main/java/feast/core/job/option/RunnerConfig.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+package feast.core.job.option;
+
+import feast.core.util.TypeConversion;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public abstract class RunnerConfig {
+  public String[] toArgs() throws IllegalAccessException {
```
Comment (ches) [inline core/src/main/java/feast/core/job/option/RunnerConfig.java:26]: Javadoc pls ☺️

Comment (zhilingc) [reply core/src/main/java/feast/core/job/option/RunnerConfig.java:26]: yessir

**Comment by woop** [comment]
@zhilingc is this going to be backported to 0.5?

**Pull request merged**